addons/account/models/account_tax.py

```diff
@@ -2385,11 +2386,13 @@ def _prepare_tax_lines(self, base_lines, company, tax_lines=None):
 
         # Remove tax lines having a zero amount.
         tax_lines_mapping = {
-            k: v
+            frozendict({grouping_k: k[grouping_k] for grouping_k in k if grouping_k != 'keep_zero_line'}): v
```

Comment:
```suggestion
            frozendict({grouping_k: k[grouping_k] for grouping_k in k if not grouping_k.startswith('__')}): v
```
That way, it looks less like an exception :p
Then, you rename your key to `__keep_zero_line`

Reply:
Good idea :+1: